node/subsystem-util/src/lib.rs

```diff
@@ -237,6 +237,27 @@ pub async fn signing_key_and_index(
 	None
 }
 
+/// Sign the given data with the given validator ID.
+///
+/// Returns `Ok(None)` if the private key that correponds to that validator ID is not found in the
+/// given keystore. Returns an error if the key could not be used for signing.
+pub async fn sign(
+	keystore: &SyncCryptoStorePtr,
+	key: &ValidatorId,
+	data: &[u8],
+) -> Result<Option<ValidatorSignature>, KeystoreError> {
+	use std::convert::TryInto;
+
+	let signature =
+		CryptoStore::sign_with(&**keystore, ValidatorId::ID, &key.into(), &data).await?;
+
+	match signature {
+		Some(sig) =>
+			Ok(Some(sig.try_into().map_err(|_| KeystoreError::KeyNotSupported(ValidatorId::ID))?)),
```

Comment:
Dropping the conversion error is interesting, we are dropping information here which might be valuable to diagnose what's the error.

Reply:
I should've mentioned that I just repeat the picture I found there.

https://github.com/paritytech/polkadot/blob/8952974037e854fe4de52c211802f23cb9aa0ae3/primitives/src/v1/signed.rs#L238

Lmk, if this is something you want to change

Reply:
The error is `()` anyway, so not really valuable.